src/write.rs

```diff
@@ -450,6 +465,48 @@ mod tests {
         assert_eq!(w.oplog.len(), 2);
     }
 
+    #[test]
+    fn wait_test() {
+        use std::thread;
+        use std::time::Duration;
+        let (mut w, _r) = crate::new::<i32, _>();
+
+        // Case 1: If epoch is set to default.
+        let test_epochs: crate::Epochs = Default::default();
+        let mut test_epochs = test_epochs.lock().unwrap();
+        w.wait(&mut test_epochs);
+
+        // Case 2: If one of the reader is still reading(epoch is odd and count is same as in last_epoch)
+        // and wait has been called.
+        let holded_epoch = Arc::new(AtomicUsize::new(1));
+
+        w.last_epochs = vec![2, 2, 1];
+        let mut epochs_slab = Slab::new();
+        epochs_slab.insert(Arc::new(AtomicUsize::new(2)));
+        epochs_slab.insert(Arc::new(AtomicUsize::new(2)));
+        epochs_slab.insert(Arc::clone(&holded_epoch));
+
+        // A new thread act as a reader thread that will increase epoch by 1 and break wait cycle.
+        let move_holded_epoch = Arc::clone(&holded_epoch);
+        let progress_handle = thread::spawn(move || {
+            thread::park_timeout(Duration::from_secs(2));
```

Comment:
Ah, if this is what you wanted to do, you'll probably want to use `sync::Barrier` instead so that you don't end up with a race condition in the test :)

Reply:
I think this may also end up nicer if you make the spawned thread do the call to `wait` instead. Then the main test code can first check that it's _not_ waiting before the barrier, then check that it _is_ waiting shortly after, then check that it stays that way after a call to `yield_now`, then bump the `held_epoch`, and then check that it can join the spawned thread.